### ISSUE TITLE
About the 'set_requires_grad'

### ISSUE THREAD

**Issue by csqiangwen** [junyanz/pytorch-CycleGAN-and-pix2pix#594]
Thanks for your nice work.
In your cycle_gan_model.py, I notice that before optimizing the generators, the 'requires_grad' attribution of the discriminators have been set as 'false'. But before optimizing the discriminators, there is no the same operation for the generators.
Could you please tell me the reason?

**Comment by junyanz** [comment]
It's a small speedup trick. `set_requires_grad=False` will stop calculating gradients for the discriminator while we update the generator. It can save some time and memory.

**Comment by csqiangwen** [comment]
Thank you very much!
